js-arrays/arrays.js

```diff
@@ -0,0 +1,187 @@
+function isArrayOne(a) {
+  return a instanceof Array;
+}
+
+function isArrayTwo(a) {
+  return Array.isArray(a);
+}
+
+function isArrayThree(a) {
+  return Object.prototype.toString.call(a) === "[object Array]";
+}
+
+function range(a, b, c) {
+  var arr = [];
+  if (b === undefined || b === null) {
+    if (c !== undefined) {
+      for (var i = 0; i < a; i += c) {
+        arr.push(i);
+      }
+    } else {
+      for (var i = 0; i < a; i++) {
+        arr.push(i);
+      }
+    }
+  } else if (c === undefined) {
+    for (var i = a; i < b; i++) {
+      arr.push(i);
+    }
+  } else if (c !== 0) {
+    for (var i = a; i < b; i += c) {
+      arr.push(i);
+    }
+  }
+  return arr;
+}
+
+function compactLoop(a) {
+  var arr = [];
+  if (Array.isArray(a)) {
+    for (var i = 0; i < a.length; i++) {
+      if (a[i]) {
+        arr.push(a[i]);
+      }
+    }
+    return arr;
+  } else {
+    console.log(a + " is not an array");
+  }
+}
+
+function compact(a) {
+  if (Array.isArray(a)) {
+    var arr = a.filter(function(el) {
+      return el ? true : false; 
+    });
+    return arr;
+  } else {
+    console.log(a + " is not an array");
+  }
+}
+
+function sumLoop(a) {
+  if (Array.isArray(a)) {
+    var sum = 0;
+    for (var i = 0; i < a.length; i++) {
+      sum += a[i];
+    }
+    return sum;
+  } else {
+    console.log(a + " is not an array");
+  }
+}
+
+function sum(a) {
+  if (Array.isArray(a)) {
+    return a.reduce(function(a,b) {
+      return a + b;
+    });
+  } else {
+    console.log(a + " is not an array");
+  }
+}
+
+function unique(a) {
+  if (Array.isArray(a)) {
+    var arr = [];
+    a.forEach(function(el) {
+      if (!arr.includes(el)) {
+        arr.push(el);
+      }
+    });
+    return arr;
+  } else {
+    console.log(a + " is not an array");
+  }
+}
+
+function last(a) {
+  if (Array.isArray(a)) {
+    return a[a.length-1];
+  } else {
+    console.log(a + " is not an array");
+  }
+}
+
+function excludeLast(a, b) {
+  if (Array.isArray(a)) {
+    if (b === undefined) {
+      a.pop();
```

Comment:
it's was mostly ok, but it's possible to do without mutations or creating array copy + copy mutations